src/components/Form/Form.spec.js

```diff
@@ -0,0 +1,32 @@
+import { default as Form } from ".";
+import { screen } from "@testing-library/react";
+
+describe("Search Form", () => {
+    beforeEach(() => {
+        renderWithReduxProvider(<Form />);
+    });
+
+    it("renders a heading 'GitHub Repository Search'", () => {
+        const heading = screen.getByRole("heading");
+        expect(heading).toBeInTheDocument();
+        expect(heading).toHaveTextContent("GitHub Repository Search");
+    });
+
+    it("renders a form", () => {
+        const form = screen.getByRole("search");
+        expect(form).toBeInTheDocument();
+    });
+
+    it("renders a search input", () => {
+        const input = screen.getByLabelText("Github username");
+        expect(input).toBeInTheDocument();
+        expect(input.type).toBe("search");
+        expect(input.placeholder).toBe("GitHub username");
+    });
+
+    it("renders a Search button", () => {
+        const button = screen.getByRole("button", { name: /Search/ });
+        expect(button).toBeInTheDocument();
+        expect(button.type).toBe("submit");
+    });
+});
```

Comment:
Nice tests. Any reason `handleSubmit` and `updateValue` were not tested?

Reply:
the functions are inside the local scope of the component so i didn't know how to access them, i was also having trouble getting user events to work. if you know how then please teach :pray: